W-17700088-outbound-policies-fa (#757)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/policies-included-stream-idle-timeout.adoc

= Stream Idle Timeout Policy
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images
:keywords: basic authentication, api gateway, flex gateway, gateway, policy

[width="100%", cols="5,15"]
|===
>s| Policy Name | Stream Idle Timeout

> >s|Summary  | Sets a maximum duration stream can remain idle without receiving a request or response

Reviewer: ```suggestion
>s|Summary  | Sets the maximum duration a stream can remain idle without receiving a request or response
```

>s|Category | Quality of Service
>s|First Flex Gateway version available | v1.9.0
^s| Returned Status Codes | 504 - Stream timeout
|===

== Summary

> The Stream Idle Timeout policy enables sets a maximum duration a stream can remain idle without receiving any data in either direction either the inbound (client to Flex Gateway) or outbound (Flex Gateway to upstream service) direction. This policy prevents idle connections from consuming resources unnecessarily. The timeout time starts at the last activity (request or response) on the stream. 

Reviewer: ```suggestion
The Stream Idle Timeout policy sets a maximum duration for which a stream can remain idle without receiving any data, either inbound from the client to Flex Gateway or outbound from Flex Gateway to the upstream service. The policy prevents idle connections from consuming resources unnecessarily. The timeout period starts at the last activity on the stream, whether the activity is a request or a response. 
```

> The Stream Idle Timeout policy enables you to set a maximum amount of a time a stream can remain idle without receiving any data in either direction either the inbound (client to Flex Gateway) or outbound (Flex Gateway to upstream service) direction. This policy prevents idle connections from consuming resources. The timeout time starts at the last request or response received be Flex Gateway. 

Reviewer 2: I suggest this to more clarity: The Stream Idle Timeout policy sets a maximum duration a stream can remain idle without receiving data in either direction: inbound (client to Flex Gateway) or outbound (Flex Gateway to the upstream service). This policy prevents idle streams from consuming resources unnecessarily. The timeout starts counting from the last activity (request or response data) on the stream.

> By default, all Flex Gateway API instances have a stream timeout of 300 seconds. Apply the policy overrides this default.

Reviewer: ```suggestion
By default, all Flex Gateway API instances have a stream timeout of 300 seconds. Apply the policy to override the default timeout.
```

== Configuring Policy Parameters

=== Flex Gateway Local Mode

> In Flex Gateway Local Mode, you apply the policy to your API via declarative configuration files. Refer to the following policy definition and table of parameters:

Reviewer: I recommend going through the rest of the PR and changing "the following" to "the" or "this" or another appropriate word.

Reviewer 3: Hey Tasha, I agree with this and noticed it when I wrote it. We do however use this in all the other policies. We have a policy audit coming post Flex Release, would it be ok if I just included these fixes there?

Reviewer: Sure, that makes sense!

----
- policyRef:

>     name: stream-idle-timeout

Reviewer 2: correct name is: stream-idle-timeout-flex

  config:
    timeout: number // REQUIRED
----

[%header%autowidth.spread,cols="a,a,a,a"]
|===
|Parameter | Required or Optional | Default Value | Description

| `timeout`
| Required
| 300
| Time in seconds before Flex Gateway closes the connection.

|===

==== Resource Configuration Example

----
- policyRef:

>     name: stream-idle-timeout

Reviewer 2: correct name is: stream-idle-timeout-flex

  config:
    timeout: 3
----

=== Flex Gateway Connected Mode

> When you apply the policy to your API from the UI, the following parameters are displayed:

Reviewer: Change "displayed" to a version of "show"

Reviewer 3: Same as above, used in all policies.
